pvanalytics/features/orientation.py

```diff
@@ -0,0 +1,185 @@
+
+def _freqstr_to_hours(freq):
+    # Convert pandas freqstr to hours (as a float)
+    if freq.isalpha():
+        freq = '1' + freq
+    return pd.to_timedelta(freq).seconds / 60
+
+
+def _hours(data, freq):
+    # Return the number of hours in `data` with data where timestamp
+    # spacing is given by `freq`.
+    return data.count() * _freqstr_to_hours(freq)
+
+
+def _group_by_day(data):
+    # Group data by timezone localized date.
+    #
+    # We use this function (rather than `data.resample('D')`) because
+    # Resampler.apply() makes the data tz-naive when passed to the
+    # function being applied. This causes the curve fitting functions
+    # to fail when the minute-of-the-day is used as the x-coordinate
+    # since removing timezone information can cause data for a day to
+    # span two dates.
+    #
+    # The date needs to be localized in the same timezone as
+    # data.index so we can easily reindex the series with the original
+    # index at a later time.
+    return data.groupby(
+        pd.to_datetime(data.index.date).tz_localize(data.index.tz)
+    )
+
+
+def tracking(power_or_irradiance, daytime, correlation_min=0.94,
+             fixed_max=0.96, min_hours=5, peak_min=None,
+             late_morning='08:45', early_afternoon='17:15'):
+    """Flag days where the data matches the profile of a tracking PV system.
+
+    Tracking days are identified by fitting a restricted quartic to
+    the data for each day. If the :math:`r^2` for the fit is greater
+    than `correlation_min` and the :math:`r^2` for a quadratic fit is
+    less than `fixed_max` then the data on that day is all marked
+    True. Data on days where the tracker is stuck, or there is
+    substantial variability in the data (i.e. caused by cloudiness) is
+    marked False.
+
+    Parameters
+    ----------
+    power_or_irradiance : Series
+        Timezone localized series of power or irradiance measurements.
+    daytime : Series
+        Boolean series with True for times that are during the
+        day. For best results this mask should exclude early morning
+        and evening as well as night. Morning and evening may have
+        problems with shadows that interfere with curve fitting, but
+        do not necessarily indicate that the day was not sunny.
+    correlation_min : float, default 0.94
+        Minimum :math:`r^2` for a day to be considered sunny.
+    fixed_max : float, default 0.96
+        Maximum :math:`r^2` for a quadratic fit, if the quadratic fit
+        is better than this, then tracking/fixed cannot be determined
+        and the day is marked False.
+    min_hours : float, default 5.0
+        Minimum number of hours with data to attempt a fit on a day.
+    peak_min : float, default None
+        The maximum value for a day must be greater than `peak_min`
+        for a fit to be attempted. If the maximum for a day is less
+        than `peak_min` then the day will be marked False.
+    late_morning : str, default '08:45'
```

Comment:
Not to complicate matters, but there are different ways to do this screening - on solar zenith for example. Maybe replace `late_morning` and `early_afternoon` with another Boolean series, a mask which is True where data is considered in the check for stuck trackers.

On that topic, I don't see where the check for stuck trackers is done. It looks like the morning/afternoon window is used to check the data for the appearance of a fixed tilt system, rather than a tracker that is mispointed and not moving (which more than likely would not have a quadratic shape). Maybe revise the comments and docstring so it doesn't appear that this function is detecting stuck trackers.

Reply:
I will change these parameters to another boolean series since it does add substantial flexibility. I kept the original behavior because I was not sure how to assign a reasonable default and I thought it might be cumbersome to always pass two day/night masks. Maybe the 'midday' mask should default to `None` and in that case `daytime` is used.

> It looks like the morning/afternoon window is used to check the data for the appearance of a fixed tilt system, rather than a tracker that is mispointed and not moving (which more than likely would not have a quadratic shape)

My understanding from the comments in the original Fleets implementation is that aggressively chopping off the morning and afternoon would result in something close enough to quadratic that the curve fitting would be somewhat successful.

Then the three criteria for whether the system is tracking come in to play:
1. quartic has a sufficiently good fit
2. quadratic has a sufficiently bad fit
3. quartic has a better fit than the quadratic

I believe that in this case it would be the 'quartic fits better than quadratic' criteria (3) that fails, rather than the 'quadratic has a sufficiently bad fit' criteria (2). One of the key points is that the quartic that we fit to the data is heavily restricted to prevent it from fitting well to anything other than the profile of a tracking PV system.

I take your point though. This is more of a test that the curve does not look like a functioning tracker than that the curve looks like a stuck tracker. I will clarify that in the documentation, and I think I should add some of the content from this comment to the documentation as well.